modules/wp-seo/wp-seo-front.php

```diff
@@ -38,9 +50,87 @@ function qtranxf_wpseo_webpage_schema( $piece, $context ) {
     add_filter( 'wpseo_breadcrumb_single_link_info', 'qtranxf_wpseo_breadcrumbs_link', 10, 3 );
     function qtranxf_wpseo_breadcrumbs_link( $link_info, $index, $crumbs ) {
         $link_info['text'] = qtranxf_useCurrentLanguageIfNotFoundUseDefaultLanguage( $link_info['text'] );
+        $link_info['url']  = qtranxf_convertURL( $link_info['url'], $lang );
 
         return $link_info;
     }
+
+    function qtranxf_wpseo_schema_website( $data ) {
+        $data['@id'] = qtranxf_convertURL( $data['@id'], $lang );
+        $data['url'] = qtranxf_convertURL( $data['url'], $lang );
+
+        return $data;
+    }
+
+    add_filter( 'wpseo_schema_website', 'qtranxf_wpseo_schema_website' );
+
+    function qtranxf_wpseo_json_ld_search_url( $search_url ) {
+        $search_url = qtranxf_convertURL( $search_url, $lang );
+
+        return $search_url;
+    }
+
+    add_filter( 'wpseo_json_ld_search_url', 'qtranxf_wpseo_json_ld_search_url', 10, 3 );
+
+    function qtranxf_wpseo_schema_imageobject( $data ) {
+        $data['@id'] = qtranxf_convertURL( $data['@id'], $lang );
+
+        return $data;
+    }
+
+    add_filter( 'wpseo_schema_imageobject', 'qtranxf_wpseo_schema_imageobject' );
+
+    function qtranxf_wpseo_schema_webpage( $data ) {
+        $data['@id'] = qtranxf_convertURL( $data['@id'], $lang );
+        $data['url'] = qtranxf_convertURL( $data['url'], $lang );
+        if ( $data['isPartOf']['@id'] ) {
+            $data['isPartOf']['@id'] = qtranxf_convertURL( $data['isPartOf']['@id'], $lang );
+        }
+        if ( $data['primaryImageOfPage']['@id'] ) {
+            $data['primaryImageOfPage']['@id'] = qtranxf_convertURL( $data['primaryImageOfPage']['@id'], $lang );
+        }
+        //if($data['author']['@id']){$data['author']['@id'] = qtranxf_convertURL($data['author']['@id'], $lang);} //Not sure is it required to filter or not???
```

Comment:
Probably needed for consistency.

Reply:
I've commented this line because I didn't understand how Yoast plugin output schema in frontend.

For example for webpage schema - items like "$data['@id']" and "$data['url']" have wrong urls without using a filter "qtranxf_convertURL". And for item "$data['author']['@id']" url is correct and points to the current language.

Same situation for all other types of schema - they all work correct except those that was filtered with "qtranxf_convertURL" function in "wp-seo-front.php".

Thoughts:
Looks like when Yoast plugin outputs schema in some cases they use URLs that is placed in "indexables" table - and that's why URLs are wrong in different languages and need to be filtered. And in other cases their plugin get URLs using default wordpress tables.

Reply:
OK I leave it commented for now.